drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,432 @@
+	}
+
+	return ret;
+}
+
+static int lis2dux12_set_range(const struct device *dev, uint8_t range)
+{
+	int err = 0;
+	struct lis2dux12_data *data = dev->data;
+
+	switch (range) {
+	default:
+	case 0U:
+		data->gain = lis2dux12_from_fs2g_to_mg(1);
+		break;
+	case 1U:
+		data->gain = lis2dux12_from_fs4g_to_mg(1);
+		break;
+	case 2U:
+		data->gain = lis2dux12_from_fs8g_to_mg(1);
+		break;
+	case 3U:
+		data->gain = lis2dux12_from_fs16g_to_mg(1);
+		break;
+	}
+
+	return err;
+}
+
+static int lis2dux12_accel_config(const struct device *dev,
+				 enum sensor_channel chan,
+				 enum sensor_attribute attr,
+				 const struct sensor_value *val)
+{
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		return lis2dux12_set_range(dev, sensor_ms2_to_g(val));
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		LOG_DBG("%s: set odr to %d Hz", dev->name, val->val1);
+		return lis2dux12_set_odr(dev, LIS2DUX12_ODR_TO_REG(val->val1));
+	default:
+		LOG_DBG("Accel attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+
+static int lis2dux12_attr_set(const struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     const struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		return lis2dux12_accel_config(dev, chan, attr, val);
+	default:
+		LOG_WRN("attr_set() not supported on this channel.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lis2dux12_sample_fetch_accel(const struct device *dev)
+{
+	struct lis2dux12_data *data = dev->data;
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	/* fetch raw data sample */
+	lis2dux12_md_t mode = {
+					.fs = cfg->range
+	};
+	lis2dux12_xl_data_t xzy_data = { 0 };
+	if (lis2dux12_xl_data_get(ctx, &mode, &xzy_data) < 0) {
+		LOG_ERR("Failed to fetch raw data sample");
+		return -EIO;
+	}
+
+	data->sample_x = sys_le16_to_cpu(xzy_data.raw[0]);
+	data->sample_y = sys_le16_to_cpu(xzy_data.raw[1]);
+	data->sample_z = sys_le16_to_cpu(xzy_data.raw[2]);
+
+	return 0;
+}
+
+#ifdef CONFIG_LIS2DUX12_ENABLE_TEMP
+static int lis2dux12_sample_fetch_temp(const struct device *dev)
+{
+	struct lis2dux12_data *data = dev->data;
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	uint8_t buf[6];
+
+	/* fetch raw data sample */
+	lis2dux12_md_t mode;
+	lis2dux12_outt_data_t temp_data = { 0 };
+	if (lis2dux12_outt_data_get(ctx, &mode, &temp_data) < 0) {
+			LOG_ERR("Failed to fetch raw temperature data sample");
+			return -EIO;
+	}
+
+	data->sample_temp = sys_le16_to_cpu(temp_data.raw[0]);
+
+	return 0;
+}
+#endif
+
+static int lis2dux12_sample_fetch(const struct device *dev,
+				 enum sensor_channel chan)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		lis2dux12_sample_fetch_accel(dev);
+		break;
+#if defined(CONFIG_LIS2DUX12_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		/* ToDo:
+		lis2dux12_sample_fetch_temp(dev)
+		*/
```

Comment:
what's up with doing just `lis2dux12_sample_fetch_temp`?

Reply:
Thanks for catching that.